mm-go-irckit/server_commands.go

```diff
@@ -384,6 +384,10 @@ func CmdPrivMsg(s Server, u *User, msg *irc.Message) error {
 		msgID, err2 := u.br.MsgChannel(ch.ID(), msg.Trailing)
 		if err2 != nil {
 			u.MsgSpoofUser(u, u.br.Protocol(), "msg: "+msg.Trailing+" could not be send: "+err2.Error())
+		} else {
+			u.msgLastMutex.Lock()
+			defer u.msgLastMutex.Unlock()
+			u.msgLast[ch.ID()] = msgID
```

Comment:
```suggestion
			return err2
		}
		
		u.msgLastMutex.Lock()
		defer u.msgLastMutex.Unlock()
		
		u.msgLast[ch.ID()] = msgID
```

Else isn't needed there, we should return first (not your fault, I should've done that)